### ISSUE TITLE
plan to promote to std

### ISSUE THREAD

**Issue by lun-4** [lun-4/zigdig#10]
this has been the plan but while i have a library that can let you write a heapless dns client, i do not know if its design actually makes _sense_ for std

so, until then, i am leaving this issue open to see if users of the library want to provide feedback for its api design, and if nothing happens in a while, maybe i'll propose another PR

**Comment by lun-4** [comment]
3 months have passed and I don't think anyone approached me with suggestions, so I'll take another pass of my code and start integrating it into an std PR

**Comment by lun-4** [comment]
note to self: I need to implement this part of stdlib as well if I want to replace the entire lookup chain: https://github.com/ziglang/zig/blob/7185ccad14068729cc805214aebd3bac62e99c73/lib/std/net.zig#L1024

**Comment by lun-4** [comment]
I was able to implement
- linuxLookupName(), with RFC 6761 Section 6.3.3 and _some parts of_ RFC 6724
- linuxLookupNameFromHosts(), basically a copy-paste from std.net as that already works fine

missing
- finding out what's missing between std.net RFC 6724 and mine
- query nameservers in parallel via poll(): https://github.com/ziglang/zig/blob/ec60156f187a9baa02e0ac7e15ee7a864f098dcf/lib/std/net.zig#L1659
